sim/battle-actions.ts

```diff
@@ -1764,8 +1764,9 @@ export class BattleActions {
 			// then the Stellar tera type applies a one-time 2x STAB boost for that type,
 			// and then goes back to using the regular 1.5x STAB boost for those types.
 			if (pokemon.terastallized === 'Stellar') {
-				if (!pokemon.stellarBoostedTypes.includes(type)) {
+				if (!pokemon.stellarBoostedTypes.includes(type) || move.stellarBoosted) {
 					stab = isSTAB ? 2 : [4915, 4096];
+					move.stellarBoosted = true;
 					if (pokemon.species.name !== 'Terapagos-Stellar') {
 						pokemon.stellarBoostedTypes.push(type);
```

Comment:
Technically this will cause the type to be saved for each hit of the move but I doubt that's a major concern.

Reply:
hmm yeah i didn't think about that. it's only a couple more instances on each type and only for multihit moves, so it should be largely inconsequential in terms of extra memory usage.